crates/egui/src/data/input.rs

```diff
@@ -7,6 +7,27 @@ use crate::{
     Key, Theme, ViewportId, ViewportIdMap,
 };
 
+/// Safe area insets of the screen. I.e., area not covered by the status bar, navigation bar, etc.
+#[derive(Debug, PartialEq, Copy, Clone, Default)]
+#[cfg_attr(feature = "serde", derive(serde::Deserialize, serde::Serialize))]
+pub struct SafeArea {
+    pub top: f32,
+    pub right: f32,
+    pub bottom: f32,
+    pub left: f32,
```

Comment:
This needs better docstring.

Does the struct describe the bounds of the safe area? If so, why isn't it a `Rect`?

Or is this describing the outer margins by which you need to shrink in order to make the screen rect safe? If so, can  we use `MarginF32` ?